gluster-exporter/metric_brick.go

```diff
@@ -128,6 +179,160 @@ func diskUsage(path string) (disk DiskStatus, err error) {
+	Attr            string
+	Size            float64
+	Path            string
+	MetadataSize    float64
+	MetadataPercent float64
+	VGName          string
+}
+
+func getLVS() ([]LVMStat, error) {
+	cmd := "lvm vgs --unquoted --noheading --nameprefixes --separator $ --nosuffix --units m -o lv_uuid,lv_name,data_percent,pool_lv,lv_attr,lv_size,lv_path,lv_metadata_size,metadata_percent,vg_name"
+
+	out, err := exec.Command("sh", "-c", cmd).Output()
+	lvmDet := []LVMStat{}
+	if err != nil {
+		// TODO: log the error
+		return lvmDet, err
+	}
+	for _, line := range strings.Split(string(out), "\n") {
+		if line != "" {
+			var obj LVMStat
+			for _, value := range strings.Split(line, "$") {
+				arr := strings.Split(value, "=")
+				switch arr[0] {
+				case "LVM2_LV_UUID":
+					obj.UUID = arr[1]
+				case "LVM2_LV_NAME":
+					obj.Name = arr[1]
+				case "LVM2_DATA_PERCENT":
+					obj.DataPercent, err = strconv.ParseFloat(arr[1], 64)
+					if err != nil {
+						// TODO: log the error as parsing of data failed
+						return return lvmDet, err
+					}
+				case "LVM2_POOL_LV":
+					obj.PoolLV = arr[1]
+				case "LVM2_LV_ATTR":
+					obj.Attr = arr[1]
+				case "LVM2_LV_SIZE":
+					obj.Size, err = strconv.ParseFloat(arr[1], 64)
+					if err != nil {
+						// TODO: log the error as parsing of data failed
+						return return lvmDet, err
+					}
+				case "LVM2_LV_PATH":
+					obj.Path = arr[1]
+				case "LVM2_LV_METADATA_SIZE":
+					obj.MetadataSize, err = strconv.ParseFloat(arr[1], 64)
+					if err != nil {
+						// TODO: log the error as parsing of data failed
+						return return lvmDet, err
+					}
+				case "LVM2_METADATA_PERCENT":
+					obj.MetadataPercent, err = strconv.ParseFloat(arr[1], 64)
+					if err != nil {
+						// TODO: log the error as parsing of data failed
+						return return lvmDet, err
+					}
+				case "LVM2_VG_NAME":
+					obj.VGName = arr[1]
+				default:
+					// Not interested
+				}
+			}
+			if obj.Attr[0] == 't' {
+				obj.Device = fmt.Sprintf("%s/%s", obj.VGName, obj.Name)
+			} else {
+				obj.Device, err = getRealPath(obj.Path)
+				if err != nil {
+					// TODO: log the error
+					return return lvmDet, err
+				}
+			}
+			lvmDet = append(lvmDet, obj)
+		}
+	}
+	return lvmDet, nil
+}
+
+// ProcMounts represents list of items from /proc/mounts
+type ProcMounts struct {
+	Name         string
+	Device       string
+	FSType       string
+	MountOptions string
+}
+
+func parseProcMounts() ([]ProcMounts, error) {
+	procMounts := []ProcMounts{}
+	b, err := ioutil.ReadFile("/proc/mounts")
+	if err != nil {
+		return procMounts, err
+	}
+	for _, line := range strings.Split(string(b), "\n") {
+		if strings.HasPrefix(line, "/") {
+			tokens := strings.Fields(line)
+			procMounts = append(procMounts,
+				ProcMounts{Name: tokens[1], Device: tokens[0], FSType: tokens[2], MountOptions: tokens[3]})
+		}
+	}
+	return procMounts, nil
+}
+
+func getRealPath(path string) (string, error) {
+	cmd := fmt.Sprintf("realpath %s", path)
```

Comment:
Can this be https://golang.org/pkg/path/filepath/#Abs or some variant thereof instead of calling out?

Reply:
I feel yes. Will try to use `filepath.Abs`

Reply:
You can use `filepath.EvalSymlinks` (used in glusterd2 https://github.com/gluster/glusterd2/blob/master/glusterd2/volume/volume-utils.go#L158)

Reply:
Ack. Will change and update the PR.